packages/cerebral-provider-firebase/src/createOnValue.js

```diff
@@ -4,16 +4,32 @@ import {
 } from './helpers'
 
 export default function createOnValue (controller) {
-  return (path, signal, options) => {
+  return (path, signal, options = {}) => {
+    let hasEmittedInitialValue = false
     listenTo(
       createRef(path, options),
       path,
       'value',
       signal,
       (data) => {
-        controller.getSignal(signal)({
+        if (!hasEmittedInitialValue) {
+          hasEmittedInitialValue = true
+          return
+        }
+
+        const initialPayload = {
           value: data.val()
-        })
+        }
+        let payload = initialPayload
+
+        if (options.payload) {
+          payload = Object.keys(options.payload).reduce((payload, key) => {
+            payload[key] = options.payload[key]
+
+            return payload
+          }, initialPayload)
```

Comment:
Reduce does not make sense for me here and makes the code harder to understand. Why not simply use:

```js
const payload = Object.assign({}, {value: data.val()}, options.payload)
```